### ISSUE TITLE
Support querystring array parameters

### ISSUE THREAD

**Issue by cdidyk** [jpodwys/superagent-cache-plugin#13]
I've been using superagent-cache-plugin for awhile now and it generally works great! However, I ran into an issue today that I think is worth raising. If the querystring includes something like `?colors[]=orange&colors[]=blue`, the key that superagent-cache-plugin generates is `{..., params: { 'colors[]': 'blue' } }`. This means that any request that ends in `colors[]=blue` (e.g. `?colors[]=blue`, `?colors[]=green&colors[]=yellow&colors[]=blue`) will incorrectly all return the same cached response.

I propose that the arrayToObj function in utils be changed as follows:

```es6
function(arr) {
  if (arr && arr.length) {
    var obj = {};
    for (var i = 0; i < arr.length; i++) {
      var str = arr[i];
      var kvArray = str.split('&');
      for (var j = 0; j < kvArray.length; j++) {
        var kvString = kvArray[j].split('=');
        if (obj[kvString[0]]) {
          obj[kvString[0]] = obj[kvString[0]] + ',' + kvString[1];
        } else {
          obj[kvString[0]] = kvString[1];
        }
      }
    }
    return obj;
  }
  return null;
}
```

Alternatively, you might consider exposing a way for consumers to supply their own custom keygen function as part of the config options supplied to the constructor. However, this seems to not be inline with the current API.

What do you think?

(I intended to submit this as a pull request, but I can't get the tests to run locally for me. They run and complete without any output, regardless of whether they should all pass or not.)

**Comment by jpodwys** [comment]
Thank you for finding and addressing this! I've confirmed that the tests are hanging on the master branch. The master branch houses the 3.x release changes. I should have kept those changes in the 3.x branch until release.

I've made a minor update to the 2.x branch that gets tests running (they still hang but only after all tests have completed). Please make your PR to the 2.x branch. Once merged, I'll cut a new 2.x release.

**Comment by cdidyk** [comment]
@jpodwys Thanks, that update made me able to test. I made the PR to the 2.x branch as requested. Let me know if there are any changes you'd like me to make.

**Comment by jpodwys** [comment]
Thanks for the PR! Published to NPM as `2.1.1`.
